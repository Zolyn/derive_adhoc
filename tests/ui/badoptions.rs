// Tests involving driver and template options

use derive_adhoc::{define_derive_adhoc, derive_adhoc, Adhoc};

define_derive_adhoc! {
    BadOptionsTemplate =
    broken template;
}

#[derive(Adhoc)]
#[derive_adhoc(BadOptionsTemplate[dbg])]
struct BadOptionsDriver;

derive_adhoc! {
    BadOptionsDriver with unknown option:
}

derive_adhoc! {
    BadOptionsDriver for wombat:
}

derive_adhoc! {
    BadOptionsDriver for union:
}

derive_adhoc! {
    BadOptionsDriver for struct:
}

derive_adhoc! {
    BadOptionsDriver dbg:
    syntax error;
}

derive_adhoc! {
    BadOptionsDriver dbg, for struct, for union:
}

fn main() {}
