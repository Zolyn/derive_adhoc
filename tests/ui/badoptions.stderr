error: unknown derive-adhoc option
 --> ui/badoptions.rs:8:14
  |
8 |     DataType with unknown option:
  |              ^^^^

error: unknown value for `for`
  --> ui/badoptions.rs:12:18
   |
12 |     DataType for wombat:
   |                  ^^^^^^

error: expected driver kind union, but driver is struct (expected kind)
  --> ui/badoptions.rs:16:18
   |
16 |     DataType for union:
   |                  ^^^^^

error: expected driver kind union, but driver is struct (actual kind)
  --> ui/badoptions.rs:5:1
   |
5  |   struct DataType;
   |   ^^^^^^^^^^^^^^^^
...
15 | / derive_adhoc! {
16 | |     DataType for union:
17 | | }
   | |_- in this macro invocation
   |
   = note: this error originates in the macro `derive_adhoc_driver_DataType` which comes from the expansion of the macro `derive_adhoc` (in Nightly builds, run with -Z macro-backtrace for more info)
